core/metadata.cpp

```diff
@@ -6,22 +6,38 @@
 #include <QFile>
 #include <QDateTime>
 
-// Fetch quint16 in big endian mode from QFile and return 0 on error.
-// This is a very specialized function for parsing JPEGs, therefore we can get away with such an in-band error code.
-static inline quint16 getShortBE(QFile &f)
+// Fetch unsigned integer of arbitrary length from file or memory in big-endian order.
+// On file error, return default value.
+// This is hackish, but it works.
+// Note: only works for unsigned integer types. This is not tested for!
```

Comment:
"This is not tested for"... what...? those last two lines of comment combined make me think that there's room to improve at least this explanation. And if there are limitations here, they should be explained in the commit message (which is lacking detail, but stupidly one cannot comment on commit messages in PR reviews)

I see the elegance in this solution - the overloading of the same function name with different arguments and using the template to allow different "width" of arguments saves a lot of writing. It still gives me pause when I read the code and I had to edit this comment twice when I thought I found something wrong and then realized that this was more clever than I first thought... but I guess that's mostly just a sign of my limitations, not an issue with the code.

So much as this doesn't come "naturally" to me, I think this is smart use of C++ to make our code more robust.

Reply:
Yeah, sorry for the non-existing commit message - it was very late. I will try to come up with one.

"This is not tested for" -> The template doesn't test whether it is instantiated with an unsigned type. I will improve the comments.

Reply:
@bstoeger - to repeat something I've said a few times. I really love that we have new developers with great ideas who contribute. I know that "it was very late" problem all too well.
I always feel bad when I push back on PRs because of commit messages or comments in the code... but I think especially for the parts of the code that are stretching my comfort zone (as far as C++ knowledge goes, for example) I'd be a fool if I didn't try to pay attention to this. Otherwise I'll regret this deeply a year or three from now...

Reply:
No worries - after all that's the whole point of an "RFC".